index.js

```diff
@@ -1,9 +1,12 @@
 'use strict';
+var fs = require('fs');
 var path = require('path');
 var childProcess = require('child_process');
-var xdgEmptyTrash = require('xdg-empty-trash');
 var runApplescript = require('run-applescript');
 var pify = require('pify');
+var Promise = require('pinkie-promise');
```

Comment:
You can drop the polyfill usage. Using this module as an API is such an edge I really don't care about Node 0.10 compat.